src/test/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/ParseIcebergColumnTreeTest.java

```diff
@@ -0,0 +1,205 @@
+                + "    {\"id\":26,\"name\":\"nested_key1\",\"required\":false,\"type\":\"string\"},"
+                + "    {\"id\":27,\"name\":\"nested_key2\",\"required\":false,\"type\":\"string\"}"
+                + "]}}]}",
+            "OBJECT(k1 NUMBER(10,0), k2 NUMBER(10,0), nested_object"
+                + " OBJECT(nested_key1 VARCHAR(16777216), nested_key2 VARCHAR(16777216)))"),
+        arguments(
+            "{\"type\":\"struct\",\"fields\":[{\"id\":2,\"name\":\"offset\",\"required\":false,\"type\":\"int\"},{\"id\":3,\"name\":\"topic\",\"required\":false,\"type\":\"string\"},{\"id\":4,\"name\":\"partition\",\"required\":false,\"type\":\"int\"},{\"id\":5,\"name\":\"key\",\"required\":false,\"type\":\"string\"},{\"id\":6,\"name\":\"schema_id\",\"required\":false,\"type\":\"int\"},{\"id\":7,\"name\":\"key_schema_id\",\"required\":false,\"type\":\"int\"},{\"id\":8,\"name\":\"CreateTime\",\"required\":false,\"type\":\"long\"},{\"id\":9,\"name\":\"LogAppendTime\",\"required\":false,\"type\":\"long\"},{\"id\":10,\"name\":\"SnowflakeConnectorPushTime\",\"required\":false,\"type\":\"long\"},{\"id\":11,\"name\":\"headers\",\"required\":false,\"type\":{\"type\":\"map\",\"key-id\":12,\"key\":\"string\",\"value-id\":13,\"value\":\"string\",\"value-required\":false}}]}\n",
+            "OBJECT(offset NUMBER(10,0), topic VARCHAR(16777216), partition"
+                + " NUMBER(10,0), key VARCHAR(16777216), schema_id NUMBER(10,0), key_schema_id"
+                + " NUMBER(10,0), CreateTime LONG, LogAppendTime LONG,"
+                + " SnowflakeConnectorPushTime LONG, headers MAP(VARCHAR(16777216),"
+                + " VARCHAR(16777216)))"));
+  }
+
+  @ParameterizedTest
+  @MethodSource("parseFromJsonArguments")
+  void parseFromJsonRecordSchema(String jsonString, String expectedType) {
+    // given
+    SinkRecord record = createKafkaRecord(jsonString, false);
+    JsonNode recordNode = RecordService.convertToJson(record.valueSchema(), record.value(), true);
+    IcebergColumnJsonValuePair columnValuePair =
+        IcebergColumnJsonValuePair.from(recordNode.fields().next());
+    // when
+    IcebergColumnTree tree = new IcebergColumnTree(columnValuePair);
+    // then
+    Assertions.assertEquals(expectedType, tree.buildType());
+    Assertions.assertEquals("TESTCOLUMNNAME", tree.getColumnName());
+  }
+
+  static Stream<Arguments> parseFromJsonArguments() {
+    return Stream.of(
+        arguments("{\"testColumnName\" : 1 }", "LONG"),
+        arguments(
+            "{ \"testColumnName\": {" + "\"k1\" : 1," + "\"k2\" : 2" + "} " + "}",
+            "OBJECT(k1 LONG, k2 LONG)"),
+        arguments(
+            "{ \"testColumnName\": {"
+                + "\"k1\" : { \"nested_key1\" : 1},"
+                + "\"k2\" : { \"nested_key2\" : 2}"
+                + "}}",
+            "OBJECT(k1 OBJECT(nested_key1 LONG), k2 OBJECT(nested_key2 LONG))"),
+        arguments(
```

Comment:
The order of vehicle1 and vehicle2 has changed after the parsing it's worrisome. However k1 and k2 do not change order.

Reply:
I checked it and it shouldn't affect anything.